python/cudf/tests/test_dataframe_copy.py

```diff
@@ -0,0 +1,148 @@
+    copy_deep_False_pdf = pdf.copy(deep=False)
+    copy_deep_False_pdf.iloc[1][1] = 10
+    assert pdf.iloc[1][1] == copy_deep_False_pdf.iloc[1][1]
+    copy_deep_False_pdf['b'] = [0, 0, 0]
+    assert np.array_equal(pdf['b'].values, copy_deep_False_pdf['b'].values)
+
+
+def test_pandas_dataframe_copy_deep_True():
+    pdf = pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+                       columns=['a', 'b', 'c'])
+    copy_deep_True_pdf = pdf.copy(deep=True)
+    copy_deep_True_pdf.iloc[1][1] = 10
+    assert pdf.iloc[1][1] != copy_deep_True_pdf.iloc[1][1]
+    copy_deep_True_pdf['b'] = [0, 0, 0]
+    assert not np.array_equal(pdf['b'].values, copy_deep_True_pdf['b'].values)
+
+
+def test_pandas_dataframe_deepcopy():
+    pdf = pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+                       columns=['a', 'b', 'c'])
+    from copy import deepcopy
+    deepcopy_pdf = deepcopy(pdf)
+    deepcopy_pdf.iloc[1][1] = 10
+    assert pdf.iloc[1][1] != deepcopy_pdf.iloc[1][1]
+    deepcopy_pdf['b'] = [0, 0, 0]
+    assert not np.array_equal(pdf['b'].values, deepcopy_pdf['b'].values)
+
+
+def test_cudf_dataframe_copy():
+    cdf = DataFrame.from_pandas(pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+                                             columns=['a', 'b', 'c']))
+    copy_deep_False_cdf = cdf.copy()
+    copy_deep_False_cdf['b'] = [0, 0, 0]
+    assert not np.array_equal(cdf['b'].to_array(), copy_deep_False_cdf['b']
+                          .to_array())
+
+
+def test_cudf_dataframe_copy_deep_False():
+    cdf = DataFrame.from_pandas(pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+                                             columns=['a', 'b', 'c']))
+    copy_deep_False_cdf = cdf.copy(deep=False)
+    copy_deep_False_cdf['b'] = [0, 0, 0]
+    assert np.array_equal(cdf['b'].to_array(), copy_deep_False_cdf['b']
+                          .to_array())
+
+
+def test_cudf_dataframe_copy_deep_True():
+    cdf = DataFrame.from_pandas(pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+                                             columns=['a', 'b', 'c']))
+    copy_deep_True_cdf = cdf.copy(deep=True)
+    copy_deep_True_cdf['b'] = [0, 0, 0]
+    assert not np.array_equal(cdf['b'].to_array(), copy_deep_True_cdf['b']
+                              .to_array())
+
+
+def test_cudf_dataframe_deepcopy():
+    cdf = DataFrame.from_pandas(pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+                                             columns=['a', 'b', 'c']))
+    from copy import deepcopy
+    deepcopy_cdf = deepcopy(cdf)
+    deepcopy_cdf['b'] = [0, 0, 0]
+    assert not np.array_equal(
+        cdf['b'].to_array(), deepcopy_cdf['b'].to_array())
+
+
+"""
+DataFrame copy bounds checking - sizes 0 through 10 perform as expected
+"""
+
+
+@pytest.mark.parametrize('ncols', [0, 1, 2, 10])
+@pytest.mark.parametrize(
+    'data_type',
+    ['int8', 'int16', 'int32', 'int64', 'float32', 'float64', 'datetime64[ms]']
```

Comment:
Any reason you didn't include 'category'?

Reply:
Copy/pasto, this pattern exists in `dataframe.py`. Should we add category to the others, do you think?

Reply:
Likely copy pasted from a function that inherently can't support categoricals. I'd add it into here as categories are unique in that it has a host side object for the category values that would need to be copied as well.

Reply:
I'd like to see a 'category' dtype test either in this test or as a separate test if it's more convenient.